velox/connectors/hive/iceberg/tests/IcebergReadTest.cpp

```diff
@@ -834,6 +834,323 @@ TEST_F(HiveIcebergTest, testPartitionedRead) {
   HiveConnectorTestBase::assertQuery(plan, splits, "SELECT 0, '2018-04-06'");
 }
 
+TEST_F(HiveIcebergTest, testReadDecimal) {
+  RowTypePtr rowType{ROW({"c0", "price"}, {BIGINT(), DECIMAL(10, 2)})};
+  std::unordered_map<std::string, std::optional<std::string>> partitionKeys;
+
+  std::vector<std::shared_ptr<ConnectorSplit>> splits;
+  std::vector<RowVectorPtr> allDataVectors;
+  std::vector<std::shared_ptr<TempFilePath>> tempFiles;
+
+  std::vector<std::string> decimalPartitionValues = {"123.45", "678.90"};
+
+  for (auto i = 0; i < decimalPartitionValues.size(); ++i) {
+    std::vector<RowVectorPtr> dataVectors;
+    // Create data file with only c0 column (price is partition column).
+    VectorPtr c0 = makeFlatVector<int64_t>({i * 10, i * 10 + 1, i * 10 + 2});
+    dataVectors.push_back(makeRowVector({"c0"}, {c0}));
+
+    auto dataFilePath = TempFilePath::create();
+    writeToFile(
+        dataFilePath->getPath(), dataVectors, config_, flushPolicyFactory_);
+
+    partitionKeys["price"] = decimalPartitionValues[i];
+    auto icebergSplits =
+        makeIcebergSplits(dataFilePath->getPath(), {}, partitionKeys);
+    splits.insert(splits.end(), icebergSplits.begin(), icebergSplits.end());
+    tempFiles.push_back(dataFilePath);
+
+    VectorPtr expectedPrice =
+        makeFlatVector<int64_t>({12345, 12345, 12345}, DECIMAL(10, 2));
+    if (i == 1) {
+      expectedPrice =
+          makeFlatVector<int64_t>({67890, 67890, 67890}, DECIMAL(10, 2));
+    }
+    allDataVectors.push_back(
+        makeRowVector({"c0", "price"}, {c0, expectedPrice}));
+  }
+
+  ColumnHandleMap assignments;
+  assignments.insert(
+      {"c0",
+       std::make_shared<HiveColumnHandle>(
+           "c0",
+           HiveColumnHandle::ColumnType::kRegular,
+           rowType->childAt(0),
+           rowType->childAt(0))});
```

Comment:
Add helper method to reduce copy-paste. Updating existing code in this file as well.

```
makeColumnHandle("c0", rowType->childAt(0));
makePartitionKeyHandle("price", rowType->childAt(1));
```